src/detect-engine-tag.c

```diff
@@ -1315,53 +1223,37 @@ static int DetectTagTestPacket06 (void)
         p[i]->flow->protoctx = &ssn;
         SigMatchSignatures(&th_v, de_ctx, det_ctx, p[i]);
 
-        if (UTHCheckPacketMatchResults(p[i], sid, (uint32_t *)&results[i][0], numsigs) == 0)
-            goto cleanup;
+        FAIL_IF(UTHCheckPacketMatchResults(p[i], sid, (uint32_t *)&results[i][0], numsigs) == 0);
 
         SCLogDebug("packet %d flag %s", i, p[i]->flags & PKT_HAS_TAG ? "true" : "false");
 
         /* see if the PKT_HAS_TAG is set on the packet if needed */
-        int expect;
-        if (i == 0 || i == 3 || i == 4 || i == 5 || i == 6)
-            expect = FALSE;
-        else
-            expect = TRUE;
-        if (((p[i]->flags & PKT_HAS_TAG) ? TRUE : FALSE) != expect)
-            goto cleanup;
+        int expect = (i == 0 || i == 3 || i == 4 || i == 5 || i == 6);
```

Comment:
expect is now the reverse of before

btw it would be nice to convert it to `bool` like in the other tests above